phoenix-core/src/main/java/org/apache/phoenix/schema/stats/GuidePostsInfo.java

```diff
@@ -53,22 +70,21 @@
      * @param guidePosts
      * @param rowCount
      */
-    public GuidePostsInfo(long byteCount, List<byte[]> guidePosts, long rowCount) {
-        this.guidePosts = ImmutableList.copyOf(guidePosts);
-        int size = 0;
-        for (byte[] key : guidePosts) {
-            size += key.length;
-        }
-        this.keyByteSize = size;
+    public GuidePostsInfo(long byteCount, ImmutableBytesWritable guidePosts, long rowCount, int maxLength, int guidePostsCount) {
+        this.guidePosts = guidePosts;
```

Comment:
Best to make a copy of the ImmutableBytesWritable here in case the caller uses the ptr again. Note that this doesn't copy the underlying bytes:

    this.guidePosts = new ImmutableBytesWritable(guidePosts);